### DISCUSSION TITLE
Parameters to tweak in order to detect/recognize small images or croped images

### DISCUSSION THREAD

**Discussion by akashlp27** [mindee/doctr#1021] [Q&A]
Hey there, I have a question, are there any parameters in the pipeline, wherein we can alter in order to get the better predictions, I was able to detect the attached image using "craft" but not with "db_res50/doctr" , or any approaches in order to get better predictions would help a lot, thanks.

![op2](https://user-images.githubusercontent.com/52736048/184849287-b19012d2-5444-46fe-b39e-fc14933d77dd.jpg)

**Comment by frgfm** [comment]
Hello @akashlp27 :wave:

I'm not sure I fully understand your request:
- only in docTR, which task are you trying to achieve? (text detection outside of docTR & text recognition with docTR?)
- you refer to better predictions, please specify what you want to get for your example image and what you get using only docTR on end-to-end :pray: 

Cheers :v:

**Reply by akashlp27** [reply]
Hey @frgfm , thanks for replying, only in docTR I was trying to run the entire pipeline using the following code:

```
model = ocr_predictor(det_arch='db_resnet50', reco_arch='crnn_vgg16_bn', pretrained=True, assume_straight_pages=True)
single_img_doc = DocumentFile.from_images(os.path.join(path, folder, image))
result = model(single_img_doc)
fig = result.show(single_img_doc)
fig.show()
```

But it did not detect properly hence the recognition failed, and then when I pass the attached image to only recognition of docTR using :

```
reco_model = crnn_vgg16_bn(pretrained=True)
reco_predictor = RecognitionPredictor(PreProcessor((32, 128), preserve_aspect_ratio=True, batch_size=32,
                                                   mean=(0.694, 0.695, 0.693),
                                                   std=(0.299, 0.296, 0.301)), reco_model)
img = cv2.imread(r"C:\Users\user\Desktop\op2.jpg")
res2 = reco_predictor([img])
```

The recognition output was also not accurate and the confident is very low.

- I wanted to know that is there any thing i can vary in the docTR pipeline(both det and rec) in order to get the better output.
- btwn for the detection/recognition for such images is there any preprocessing which can be done to use with docTR

**Reply by frgfm** [reply]
Oh I can see a few problems actually:
- try to use `recognition_predictor` to avoid errors instead of rebuilding it yourself
- you're reading image with cv2, out of safety, use docTR `DocumentFile.from_images` (cv2 doesn't have the same convention as Pillow or PyTorch for images, but we harmonized everything in docTR, it's just a clean wrapper)

So first thing I'd suggest is to use this snippet instead:
```python
from doctr.models import recognition_predictor
from doctr.io import DocumentFile

# The preprocessor will be configured automatically
reco_predictor = recognition_predictor("crnn_vgg16_bn", pretrained=True)
doc = DocumentFile.from_images(r"C:\Users\user\Desktop\op2.jpg")
res2 = reco_predictor(doc)
```

If you're still having problems, let me know :)

**Reply by akashlp27** [reply]
Hey, @frgfm , thanks for the input , and for such images, the db_resnet50 of DocTr is failing any inputs on that ?

**Reply by frgfm** [reply]
Hi @akashlp27 :wave:

Would you mind providing a snippet? I'm not sure I understand your question
The snippet that I provided is only about text recognition (as it seems to be what you are interested in), but `db_resnet50` is a text detection model :thinking:
